fedimint-server/src/net/connect.rs

```diff
@@ -523,6 +530,30 @@ pub mod mock {
             }
         };
 
+        pub const INTEGRATION_TEST: StreamReliability = {
+            // Based on empirical testing: creates errors without causing tests to take
```

Comment:
This part is tricky, as changes to the code down the line might invalidate this heuristic.

In the past for similar reasons I used adaptive failure rates. Eg. time decaying failure rates, or linearly increasing with time, but geometrically slashed on each failure (kind of like TCP: https://en.wikipedia.org/wiki/Additive_increase/multiplicative_decrease) .

Reply:
> This part is tricky, as changes to the code down the line might invalidate this heuristic.

Yes, and it's okay. If tests start to hangup and no bug is found, then the parameters will be scaled down. The default parameter for day-to-day CI activities should be kept on the lower side.

But in future we can create a stress script that tries to run the tests with higher values. Then we can run it periodically (like once a day) or in ad-hoc manner. It could create fancy graphs showing how different metrics change based on different values of failure rate/latency.

> In the past for similar reasons I used adaptive failure rates. Eg. time decaying failure rates, or linearly increasing with time, but geometrically slashed on each failure (kind of like TCP: https://en.wikipedia.org/wiki/Additive_increase/multiplicative_decrease) .

An adaptive failure rate is a very interesting idea, but I think it would also be tricky to get it right. It would require keeping a state that survives between different (re)connections. For instance, you could start with a higher (or lower) failure rate then get it down (or up) after each reconnection/packet. But this may create a bias where errors are more prone to happen on some moments of the test than others. Also I guess it would still require some tuning for the increase/decrease parameters that may still need some retuning in future. You could end in the same spot but with a more complex model to deal with.